docs: add examples page (#131)
ICAIR-Sheffield/LayOpt: docs/examples.md

# Examples

## Single span

> Below is the contents of a configuration file for a single span example.

Reviewer: Not sure about the "structure" at the end, but perhaps...

```suggestion
Below is an example configuration file for a single span structure.
```

Author: No, structure is good here I think

  - [5, 2]
  - [6, 2]
  - [7, 2]
  - [8, 2]
  - [9, 2]
load_direction: # Load direction
  - 0.0
  - -1.0
load_large: 3.75 # Large load
load_small: 0.204 # Small load
max_length: 18.0 # Max length
support_points: # Support points
  - [0, 0]
  - [9, 0]
member_area_filtering: 0.001 # Fraction of maximum member area for output threshold
cvxpy:
  solver: clarabel # See <https://www.cvxpy.org/tutorial/solvers/index.html?h=solve#choosing-a-solver> for further details
filter_levels: [1.0] # List of values to filter by if empty no filtering is performed
primal_method: load_factor # Primal method
problem_name: arch_testing_smaller # Problem name
csv_filename: results.csv # Filename to save results to, if left as `results.csv` will have YYYY-MM-DD-hhmmss added
notes: smaller arch test # Notes to include
plotting:
  run: true # Whether to plot the trusses
  bar_thickness: 0.3 # Bar thickness when plotting
  dpi: 1200 # Dots per inch when plotting
```

The setup of this example is represented by the following image (produced using the
[LayOpt truss layout optimization](https://www.layopt.com/truss/) web tool):
![Single span of width 9 and height 2](img/single_span_example_setup_resized.svg)

> Saving the above configuration as `single_span_config.yaml` in your working directory, you can then run this job with

Reviewer: Perhaps overly pedantic/fussing over semantics but we don't provide an easy way for people to "save" the above file.

I can see two solutions...

1. Add an `assets` directory and include a copy of the configuration file `assets/single_span_config.yaml` and make the text link to it, advising people to right-click and save as.

```suggestion
A copy of the above configuration can be downloaded by right-clicking on [`single_span_config.yaml`](assets/single_span_config.yaml) and saving it to your working directory. You can then run this job with:
```

2. Changing the text to advise people to ...

```suggestion
Copy and paste the above text into a file called `single_span_config.yaml` and save it to your working directory. You can then run this job with:
```

Author: This is a better solution I think, I shouldn't assume that people would be able to put the yaml file together (apart from anything else, I suspect at least some users will be unfamiliar with the file type/extension).
Will resolve this and other one shortly

Author: I opted for both, with the 'copy and paste' method in parentheses.

> `layopt -c single_span_config.yaml optimise`.

Reviewer: I think having this as a code block rather than inline code would make it stand out better.

````suggestion
```shell
layopt -c single_span_config.yaml optimise.
```
````

We could also optionally include the output from running the command (at discretion we could change `log_leve: error` so that there is less output too).

Author: There's no output besides the ASCII art from `error` log level (same for warning) for these examples, and I think the `info`/default level output is too long to be worth sharing. Perhaps I could put in just the line for the final volume and maybe the output plot?

Reviewer: Yep that would be good, something so people know what the output looks like.

Do you think it would be overkill to include the `.csv` output?

Author: Hmm that could work but probably is a little overkill. Putting that on the page to test looks like it could be more confusing than just having the volume output and a plot

Author: Done this, although using `shell` code block in Markdown doesn't look quite right (odd colouring)

> 

Reviewer: Should we include an copy of the output file to show people what to expect (for both this example and the one below)?

  - [8, 0]
  - [8, 4]
load_direction: # Load direction
  - 0.0
  - -1.0
load_large: 50 # Large load
load_small: 5 # Small load
max_length: 16 # Max length
support_points: # Support points
  - [0, 0]
  - [0, 1]
  - [0, 2]
  - [0, 3]
  - [0, 4]
  - [0, 5]
  - [0, 6]
  - [0, 7] 
  - [0, 8]    
member_area_filtering: 0.001 # Fraction of maximum member area for output threshold
cvxpy:
  solver: clarabel # See <https://www.cvxpy.org/tutorial/solvers/index.html?h=solve#choosing-a-solver> for further details
filter_levels: [1.0] # List of values to filter by if empty no filtering is performed
primal_method: load_factor # Primal method
problem_name: square_cantilever # Problem name
csv_filename: results.csv # Filename to save results to, if left as `results.csv` will have YYYY-MM-DD-hhmmss added
notes: square cantilever three load points # Notes to include
plotting:
  run: true # Whether to plot the trusses
  bar_thickness: 0.3 # Bar thickness when plotting
  dpi: 1200 # Dots per inch when plotting
```

The setup of this example is represented by the following image (produced using the
[LayOpt truss layout optimization](https://www.layopt.com/truss/) web tool):

> ![Square cantilever width 8 and height 8](img/square_cantilever.png)

Reviewer: ````suggestion
![Square cantilever width 8 and height 8](img/square_cantilever.png)

Save the [`square_cantilever_config.yaml`](assets/square_cantilever_config.yaml) file to your working directory and run this with:

```shell
layopt -c square_cantilever_config.yaml optimise
```
````

And perhaps show output if you think its worthwhile.
